### PR TITLE
api reviews completed

### PR DESCRIPTION
**Pull request by markangelodabu**
Completed Reviews Api, not too confident about it though, not quite sure where getReviewsByUser or getReviewsByProducts fits in this file. Please review. Thanks

### CODE REVIEW

File: api/reviews.js [hunk 1]
```diff
@@ -1,4 +1,104 @@
...
+    getReviewsByProduct,
+    createReview,
+    updateReview,
+    deleteReview,
+} = require ("../db")
+const { requireUser } = require("./utils");
+
+reviewsRouter.get("/", async (req, res, next) => {
+    try {
+        const reviews = await  getAllReviews();
```
Comment (yeonjupark95) [inline api/reviews.js:16]: extra space between await and getAllReviews :)

File: api/reviews.js [hunk 2]
```diff
@@ -1,4 +1,104 @@
...
+        res.send(newReview)
+    }   catch (error) {
+        console.log("Error at creating a new review", error)
+        next(error);
+    }
+});
+
+reviewsRouter.get("/:userId/reviews", async (req, res, next) => {
+    const { userId } = req.params;
+    //not quite sure how to access username but the functions getReviewsByUser requires a username parameter.
```
Comment (yeonjupark95) [inline api/reviews.js:37]: Need another line to grab username:
const {username} = req.body

File: api/reviews.js [hunk 3]
```diff
@@ -1,4 +1,104 @@
...
+    try {
+        const reviewByUser = await getReviewsByUser({
+            id: userId,
```
Comment (yeonjupark95) [inline api/reviews.js:40]: username

File: api/reviews.js [hunk 4]
```diff
@@ -1,4 +1,104 @@
...
+    try {
+        const newReview = await createReview({ userId, productId, description, rating });
+        res.send(newReview)
+    }   catch (error) {
+        console.log("Error at creating a new review", error)
+        next(error);
+    }
+});
+
+reviewsRouter.get("/:userId/reviews", async (req, res, next) => {
```
Comment (yeonjupark95) [inline api/reviews.js:35]: reviewsRouter.get("/:username", async (req, res, next) => {

File: api/reviews.js (hunk 3 above)
Comment (yeonjupark95) [inline api/reviews.js:40]: username

File: api/reviews.js [hunk 5]
```diff
@@ -1,4 +1,104 @@
...
+    const { userId } = req.params;
```
Comment (yeonjupark95) [inline api/reviews.js:36]: const {username} = req.params;

**Review by lindsaynaki — APPROVED** [review]

**Pull request merged**